src/S3.ts

```diff
@@ -42,11 +42,11 @@ class S3 {
         });
     }
 
-    public async uploadBlob(key: string, data: any, eTag: string) {
+    public async uploadBlob(key: string, data: any, eTag: string, Webp: boolean) {
         const params = {
             Bucket: this.bucketName,
             Key: key,
-            Metadata: {etag: eTag },
+            Metadata: {etag: eTag ,webp: Webp},
```

Comment:
S3 should ideally not known anything about the webp problematic. No need to create dependencies to this here. Just replace with a `version` parameter taking a string value and put this as additional Metdata.

Call the function then with `webp` value in case of `webp` conversion, and `1` otherwise.